src/rgw/rgw_user.cc

```diff
@@ -2189,6 +2201,11 @@ int RGWUser::execute_modify(RGWUserAdminOpState& op_state, std::string *err_msg)
     user_info.user_email = "";
   }
 
+  // update user type, if given by user
+  if (op_state.has_user_type()) {
+    user_info.type = op_state.get_user_type();
+  }
+
```

Comment:
@rzarzynski : I tried using boost::optional for user_type:
boost::optional<RGWUserSourceType> user_type = op_state.get_user_type();
  if (user_type) {
    user_info.type = static_cast<uint32_t>(user_type);
  }

But I get compilation issues saying that user_type can not be converted from boost::optional<RGWUserSourceType> to uint32_t.

Even if I try the following:
RGWUserSourceType type = op_state.get_user_type();
I get an error saying that boost::optional<RGWUserSourceType> can't be converted to RGWUserSourceType. I understand that this error is because boost::optional enables type to have boost::None, which is not of type RGWUserSourceType, so what will be the correct way of storing user type in user_info, if I want to make use of boost::optional here?

Reply:
I guess you would need to dereference the optional:

```cpp
  boost::optional user_type = op_state.get_user_type();
  if (user_type) {
    user_info.type = *user_type;
  }
```

Reply:
Thank you, not sure why I missed that :(